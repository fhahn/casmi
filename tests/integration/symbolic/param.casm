// symbolic

function (symbolic) a: Int -> Int
function (symbolic) b: Int -> Int

rule main = {
  
  b( 1 ) := a( 1 )
  b( 2 ) := a( 2 )

  program( self ) := undef
}
init main

