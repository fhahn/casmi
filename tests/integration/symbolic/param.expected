forklog:
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,sta(1,1,sym2)).%CREATE: a(1)
tff(symbolNext, type, sym3: $int).
fof(id1,hypothesis,stb(1,1,sym3)).%CREATE: b(1)
tff(symbolNext, type, sym4: $int).
fof(id2,hypothesis,sta(1,2,sym4)).%CREATE: a(2)
tff(symbolNext, type, sym5: $int).
fof(id3,hypothesis,stb(1,2,sym5)).%CREATE: b(2)
fof(id4,hypothesis,stb(2,1,sym2)).%UPDATE: b(1)
fof(id5,hypothesis,stb(2,2,sym4)).%UPDATE: b(2)
fof(id6,hypothesis,sta(2,2,sym4)).%SYMBOLIC: a(2)
fof(id7,hypothesis,sta(2,1,sym2)).%SYMBOLIC: a(1)
fof(final0,hypothesis,sta(0,2,sym4)).%FINAL: a(2)
fof(final1,hypothesis,sta(0,1,sym2)).%FINAL: a(1)
fof(final2,hypothesis,stb(0,2,sym4)).%FINAL: b(2)
fof(final3,hypothesis,stb(0,1,sym2)).%FINAL: b(1)

